lib/Sema/MiscDiagnostics.cpp

```diff
@@ -3997,9 +3998,14 @@ void swift::fixItAccess(InFlightDiagnostic &diag, ValueDecl *VD,
     // this function is sometimes called to make access narrower, so assuming
     // that a broader scope is acceptable breaks some diagnostics.
     if (attr->getAccess() != desiredAccess) {
-      // This uses getLocation() instead of getRange() because we don't want to
-      // replace the "(set)" part of a setter attribute.
-      diag.fixItReplace(attr->getLocation(), fixItString.drop_back());
+      if (shouldNotReplace) {
+        // Remove the attribute if replacement is not preferred.
+        diag.fixItRemove(attr->getRange());
```

Comment:
Can you add a test with `public(set)`?

Reply:
Good catch! I extended the solution to also produce warnings for redundant access modifiers on setters.